jerry-core/lit/lit-char-helpers.c

```diff
@@ -289,32 +289,32 @@ lit_char_hex_to_int (ecma_char_t c) /**< code unit, corresponding to
  * @return true if decoding was successful, false otherwise
  */
 bool
-lit_read_code_point_from_hex (lit_utf8_byte_t *buf_p, /**< buffer with characters */
-                              lit_utf8_size_t number_of_characters, /**< number of characters to be read */
-                              lit_code_point_t *out_code_point_p) /**< [out] decoded result */
+lit_read_code_unit_from_hex (lit_utf8_byte_t *buf_p, /**< buffer with characters */
```

Comment:
With the help of the non-character markers (as in my previous comment), couldn't we just return a code unit from this function and compare it to U+FFFF to see whether we succeeded or not? (I'm really against out parameters given with pointers, so prefer to replace them wherever possible.)